mongo_connector/oplog_manager.py

```diff
@@ -459,6 +505,19 @@ def get_oplog_cursor(self, timestamp=None):
         If no timestamp is specified, returns a cursor to the entire oplog.
         """
         query = {'op': {'$ne': 'n'}}
+
+        if self.oplog_ns_set and not self.oplog_ex_ns_set:
+            query['ns'] = {'$in': self.oplog_ns_set}
+
+        if self.oplog_ex_ns_set and not self.oplog_ns_set:
+            query['ns'] = {'$nin': self.oplog_ex_ns_set}
+
+        if self.oplog_ns_set and self.oplog_ex_ns_set:
+            query = {'$and': [{'ns': {'$nin': self.oplog_ex_ns_set}},
+                              {'ns': {'$in': self.oplog_ns_set}}]}
+
+        LOG.debug("Cursor query: %s" % query)
```

Comment:
I am considering moving this part of logic inside the new DestMapping class. Otherwise, it needs to do regex matching which already did in that class again. And the performance will be better since inside that class there is kind of caching of the wildcard patterns.